Assignment4/question1.js

```diff
@@ -0,0 +1,70 @@
+// PART 1
+
+/*
+    1. What happens when you add a for loop/while loop/switch case block inside a function and use return instead of break? Do statements after the loop run? What is the return value? Can we pass a return value from within a loop? Can you return from inside an if block? What impact does that have?
+
+    ANSWERS:
+
+    1. What happens when you add a for loop/while loop/switch case block inside a function and use return instead of break?
+        - when return is used the flow of execution goes to parent function i.e it returns to caller function
+    
+    2. Do statements after the loop run?
+        - NO
+    
+    3. What is the return value?
+        - any type of return value can be provided or empty return can also be there
+    
+    4. Can we pass a return value from within a loop?
+        - Yes, we can pass the return value
+    
+    5. Can you return from inside an if block? What impact does that have?
+        - Yes, it makes the same impact i.e flow moves to caller(parent) function
+*/
+
+
+// PART 2
+/*
+    2. Take a function that accepts a function as a parameter (also known as callback function). 
+    function test(callback){
+    callback();
+    }
+
+    function callbackFunc(){
+    console.log(“Calling the callback function”)
+    return 5;
+    }
+    
+    Explore the following cases - what is printed in the console, and what gets returned:
+
+    test(callbackFunc)
+    test(callbackFunc())
+    test(() => callbackFunc())
+
+    What happens when you return callback() from the test function?
+    What happens when you return callback from the test function?
+
+    ANSWER:
```

Comment:
instead of writing que ans, write code and implement these scenarios and write o/p in comments